specification/policyinsights/resource-manager/Microsoft.PolicyInsights/stable/2018-04-04/policyEvents.json

```diff
@@ -0,0 +1,918 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "PolicyEventsClient",
+    "version": "2018-04-04"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/providers/{managementGroupsNamespace}/managementGroups/{managementGroupName}/providers/Microsoft.PolicyInsights/policyEvents/{policyEventsResource}/queryResults": {
+      "post": {
+        "operationId": "PolicyEvents_ListQueryResultsForManagementGroup",
+        "description": "Queries policy events for the resources under the management group.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/policyEventsResourceParameter"
+          },
+          {
+            "$ref": "#/parameters/managementGroupsNamespaceParameter"
+          },
+          {
+            "$ref": "#/parameters/managementGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/topParameter"
+          },
+          {
+            "$ref": "#/parameters/orderByParameter"
+          },
+          {
+            "$ref": "#/parameters/selectParameter"
+          },
+          {
+            "$ref": "#/parameters/fromParameter"
+          },
+          {
+            "$ref": "#/parameters/toParameter"
+          },
+          {
+            "$ref": "#/parameters/filterParameter"
+          },
+          {
+            "$ref": "#/parameters/applyParameter"
+          }
```

Comment:
What do you think about grouping parameters? [https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-grouping](https://github.com/Azure/autorest/tree/master/docs/extensions#x-ms-parameter-grouping)

Reply:
I used it. See QueryOptions

Reply:
Awesome :)

---
In reply to: [184759834](https://github.com/Azure/azure-rest-api-specs/pull/2929#discussion_r184759834) [](ancestors = 184759834)